### PR TITLE
Feat: Updated torch.nn.Modules.set_submodules()

### PR DESCRIPTION
**Pull request by Gunale0926**
modified:   torch/nn/modules/module.py

Implemented feature request by #127712.

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -665,6 +665,62 @@ def get_submodule(self, target: str) -> "Module":
...
+                fully-qualified string.)
+            module: The module to set the submodule to.
+
+        Raises:
+            AttributeError: If the target string references an invalid
+                path or resolves to something that is not an
+                ``nn.Module``
+        """
+        if target == "":
+            raise AttributeError("Cannot set the submodule without a target name!")
```
Comment (Skylion007) [inline torch/nn/modules/module.py:706]:
```suggestion
            raise ValueError("Cannot set the submodule without a target name!")
```

**Comment by Skylion007** [comment]
You need to sign the CLA. Also, there aren't any tests here.

**Review by mikaylagawarecki — APPROVED** [review]
Thanks

File: test/test_nn.py [hunk 2]
```diff
@@ -1434,6 +1434,20 @@ def test_add_module(self):
...
+        l = nn.Linear(1, 2)
+        target = "t.l"
+        net.set_submodule(target, l)
+        self.assertEqual(net.get_submodule(target), l)
+        l2 = nn.Linear(2, 1)
+        net.set_submodule(target, l2)
+        self.assertEqual(net.get_submodule(target), l2)
+        self.assertRaises(ValueError, net.set_submodule, "", l)
+        self.assertRaises(AttributeError, net.set_submodule, "a.l", l)
+        self.assertRaises(AttributeError, net.set_submodule, "t.l.l2", l2)
```
Comment (mikaylagawarecki) [inline test/test_nn.py:1449]: Hm it seems like tests are failing

**Comment by Gunale0926** [comment]
Bug fixed.

**Comment by Gunale0926** [comment]
I think some checks failed not because of my modification.

**Review by mikaylagawarecki — APPROVED** [review]

**Pull request closed without merging**